portality/templates/layouts/base.html

```diff
@@ -71,12 +71,92 @@
 
     <script type="text/javascript">
     jQuery(document).ready(function() {
-        $(".cookie-consent").on("click", setCookieConsent);
+        $(".cookie-consent #cookie-consent-hide").on("click", setCookieConsent);
         $(".site-note-dismiss").on("click", dissmissSiteNote);
     });
     </script>
 
   {% endif %}
 
+  <script>
+    // Use Feather icons
+    feather.replace()
+
+    // Responsive menu
+    var openMenu = document.querySelector(".secondary-nav__menu-toggle");
+    var nav = document.querySelector(".secondary-nav__menu");
+
+    openMenu.addEventListener('click', function() {
+      nav.classList.toggle("secondary-nav__menu-toggle--active");
+    }, false);
+
+    // Back-to-top button
+    topBtn = document.getElementById("top");
+
+    window.onscroll = function() {scrollFunction()};
+
+    function scrollFunction() {
+      if (document.body.scrollTop > 100 || document.documentElement.scrollTop > 100) {
+        topBtn.style.display = "flex";
+      } else {
+        topBtn.style.display = "none";
+      }
+    }
+
+    // When the user clicks on the button, scroll to the top of the document
+    function topFunction() {
+      document.body.scrollTop = 0; // For Safari
+      document.documentElement.scrollTop = 0; // For Chrome, Firefox, IE and Opera
+    }
+
+    // Tabs
+    $("[role='tab']").click(function(e) {
+      e.preventDefault();
+      $(this).attr("aria-selected", "true");
+      $(this).parent().siblings().children().attr("aria-selected", "false");
+      var tabpanelShow = $(this).attr("href");
+      $(tabpanelShow).attr("aria-hidden", "false");
+      $(tabpanelShow).siblings().attr("aria-hidden", "true");
+    });
+
+    // Offset when clicking on anchor link or sidenav link to compensate for fixed header
+    var headerHeight = 70;
+
+    // When on the same page
+    jQuery (function($) {
+      $('a[href*="#"]:not([href="#"]):not([role="tab"])').click(function() {
+        var target = $(this.hash);
+        $('html,body').animate({
+          scrollTop: target.offset().top - headerHeight
+        }, 50, 'linear');
+      });
+      if (location.hash){
+        var id = $(location.hash);
+      }
+
+      $(window).load(function() {
+        if (location.hash){
+          $('html,body').animate({
+            scrollTop: id.offset().top - headerHeight
+          }, 50, 'linear')
+        };
+      });
+    });
+
+    // Coming from another page
+    jQuery (document).ready (function($) {
+      var hash= window.location.hash
+      if ( hash == '' || hash == '#' || hash == undefined ) return false;
+      var target = $(hash);
+      target = target.length ? target : $('[name=' + hash.slice(1) +']');
+      if (target.length) {
+        $('html,body').animate({
+          scrollTop: target.offset().top - headerHeight //offsets for fixed header
+        }, 50, 'linear');
+      }
+    } );
+
+  </script>
+
```

Comment:
Just so you know, all this code was moved to `_js_includes.html` to keep it together with all the other base JS.

There were also a few bugfixes to it, so unless this code here has changed, and the changes need porting over to the updated version, I'd probably just drop this from the merge.

Reply:
Hmm it seems that without that JS there, some small features don’t appear though, like the "scroll to top" button: 

![1ae2413f11a8070b4a71f9c24c53bbd9](https://user-images.githubusercontent.com/6668406/93613023-34a18f00-f9c8-11ea-8d14-64b95dfb79cd.gif)

Reply:
ok, I'll have to diagnose.  In theory whether the code is there or in `_js_includes.html` shouldn't matter, it's just keeping the JS all together in the same file.  Maybe there's a loading order thing that needs looking at.

Reply:
(incidentally, and maybe or maybe not related, I get a lot of errors from the scroll code on my machine, maybe it's crashing out on some pages and stopping working.  I'll look when I get a moment)

Reply:
(I say "a lot of errors", it's always the same error, but I get it a lot :) )

Reply:
Ah, just saw that! I get the  error too. JS is definitely not my thing... 😬